Sources/Swiftfall/Swiftfall.swift

```diff
@@ -1,635 +1,637 @@
-            
-            let decoder = JSONDecoder()
-            do {
-                // Decode JSON file starting from Response struct.
-                let decoded:SetList = try decoder.decode(SetList.self, from: content)
-                completion(decoded)
-            }
-            catch {
-                do {
-                    let decoded:Error = try decoder.decode(Error.self, from: content)
-                    decoded.simplePrint()
-                }
-                catch {
-                    print("Error: \(error)")
-                    
-                    completion(nil)
-                }
-                completion(nil)
-            }
-        }
-        task.resume()
-    }
-    
-    public static func getCardList() -> CardList?
-    {
-        let call = "cards/"
-        
-        var cardlist: CardList?
-        var stop = false
-        parseCardList(call: call){
-            (newcardlist:CardList?) in
-            cardlist = newcardlist
-            stop = true
-        }
-        
-        while(!stop){
-            //Do this until parseCard is done
-        }
-        
-        return cardlist
-    }
-    
-    public static func getCardList(page:Int) -> CardList?
-    {
-        let call = "cards?page=\(page)"
-        
-        var cardlist: CardList?
-        var stop = false
-        parseCardList(call: call){
-            (newcardlist:CardList?) in
-            cardlist = newcardlist
-            stop = true
-        }
-        
-        while(!stop){
-            //Do this until parseCard is done
-        }
-        
-        return cardlist
-    }
-    
-    /// Retreives JSON data from URL and parses it with JSON decoder. Thanks Mitchell
-    static func parseCardList(call:String, completion: @escaping (CardList?) -> ()) {
-        
-        let url = URL(string: "\(scryfall)\(call)")
-        let task = URLSession.shared.dataTask(with: url!) {(data, response, error) in
-            guard let content = data else {
-                print("Error: There was no data returned from JSON file.")
-                return
-            }
-            
-            //print("\(String(data: content,encoding: .utf8))")
-            
-            let decoder = JSONDecoder()
-            do {
-                // Decode JSON file starting from Response struct.
-                let decoded:CardList = try decoder.decode(CardList.self, from: content)
-                completion(decoded)
-            }
-            catch {
-                do {
-                    let decoded:Error = try decoder.decode(Error.self, from: content)
-                    decoded.simplePrint()
-                }
-                catch {
-                    print("Error: \(error)")
-                    
-                    completion(nil)
-                }
-                completion(nil)
-            }
-        }
-        task.resume()
-    }
-    
-    
-    public static func getRulingList(code:String,number:Int) -> RulingList?
```

Comment:
This can also probably get replaced with a call to `synchronouslyGetResource(call:)` if you go that route.